fdbserver/DDTeamCollection.actor.cpp

```diff
@@ -3691,24 +3710,27 @@ bool DDTeamCollection::isWigglePausedServer(const UID& server) const {
 }
 
 std::vector<UID> DDTeamCollection::getRandomHealthyTeam(const UID& excludeServer) {
-	std::vector<int> candidates, backup;
-	for (int i = 0; i < teams.size(); ++i) {
-		if (teams[i]->isHealthy() && !teams[i]->hasServer(excludeServer)) {
-			candidates.push_back(i);
-		} else if (teams[i]->size() - (teams[i]->hasServer(excludeServer) ? 1 : 0) > 0) {
-			// If a team has at least one other server besides excludeServer, select it
-			// as a backup candidate.
-			backup.push_back(i);
+	std::vector<Reference<TCTeamInfo>> candidates, backup;
+	for (auto& teamSet : m_teamSets) {
+		const std::vector<Reference<TCTeamInfo>>& teams = teamSet->teams();
+		for (int i = 0; i < teamSet->teamCount(); ++i) {
+			if (teams[i]->isHealthy() && !teams[i]->hasServer(excludeServer)) {
+				candidates.push_back(teams[i]);
+			} else if (teams[i]->size() - (teams[i]->hasServer(excludeServer) ? 1 : 0) > 0) {
+				// If a team has at least one other server besides excludeServer, select it
+				// as a backup candidate.
+				backup.push_back(teams[i]);
+			}
```

Comment:
I have a suggestion because I've seen this pattern
```
for (auto& teamSet : m_teamSets) {
      std::vector<Reference<TCTeamInfo>> teams = teamSet->teams();
      for (auto& team : teams) {
      ....
      }
}
```
several times. Would it be possible to do something similar to `ParallelTCInfo` and `IDataDistributionTeam`?

Reply:
Yes, this makes sense. I'll do this in an incremental cleanup change.